### DISCUSSION TITLE
Log improvements

### DISCUSSION THREAD

**Discussion by stishkin** [microsoft/onefuzz#2448] [Show and tell] [answered]
Moving out data-values from log messages into CustomDimensions

![image](https://user-images.githubusercontent.com/7307755/192385697-62df532c-65e0-4807-a95c-ae22be942d05.png)

Code to produce AppInsights entry:

![image](https://user-images.githubusercontent.com/7307755/192386347-a9a7534d-3c55-4309-abd4-5ef8fb8c9be6.png)

**Comment by stishkin** [comment]
Change tags in the message from `Tag:ToState` to `{ToState}` ?

**Reply by mgreisen** [reply]
Sounds good to me.

**Comment by tevoinea** [comment]
It would also be nice if we could enforce some sort of naming consistency for these tags to make it easier to query/join across different message types. Could it be sufficient to say the tag name has to be the property name? Our models are pretty consistent.

The downside with coupling tag name to model property name is that we might want to do a custom tag name in the telemetry for whatever reason so we would need a way to override the default behavior.

**Reply by stishkin** [reply]
+1 for consistency

We have cases where tags are used but those values are not part of the models.
In the snippet that I provided we use State two times in the log message.  

For now - it's manual consistency. If we think of a method to enforce it - we'll implement that.

As we cleanup logs - there might be typos discovered in the log, but those are easy to fix.

**Comment by stishkin** [comment, marked as answer]
https://github.com/microsoft/onefuzz/pull/2450
